File: src/main/java/no/fintlabs/access/control/model/entity/ClientEntity.java
package no.fintlabs.access.control.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "client")
public class ClientEntity {
    @Id
    @Size(max = 255)
    @Column(name = "username", nullable = false)
    private String username;

    @NotNull
    @Column(name = "is_adapter", nullable = false)
    private Boolean isAdapter = false;

    @Size(max = 1000)
    @Column(name = "allowed_environments", length = 1000)
    private String allowedEnvironments;

}

File: src/main/java/no/fintlabs/access/control/model/entity/ClientEntity.kt
package no.fintlabs.access.control.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.validation.constraints.Size

@Entity
@Table(name = "client")
class ClientEntity(
    @Id
    @Column(name = "username", nullable = false)
    @field:Size(max = 255)
    val username: String,

    @Column(name = "is_adapter", nullable = false)
    val isAdapter: Boolean = false,

    @Column(name = "allowed_environments", length = 1000)
    @field:Size(max = 1000)
    val allowedEnvironments: String? = null
) {
    // JPA krever en no-args konstruktør, så vi legger til en eksplisitt
    constructor() : this(
        username = "",
        isAdapter = false,
        allowedEnvironments = null
    )
}
